sdk/servicebus/service-bus/src/connectionContext.ts

```diff
@@ -346,11 +490,34 @@ export namespace ConnectionContext {
       if (context.connection.isOpen()) {
         log.ns("Closing the amqp connection '%s' on the client.", context.connectionId);
 
-        // Close all the clients.
-        for (const id of Object.keys(context.clientContexts)) {
-          const clientContext = context.clientContexts[id];
-          await clientContext.close();
+        // Close all the senders.
+        for (const senderName of Object.keys(context.senders)) {
+          await context.senders[senderName].close();
```

Comment:
delete or clear the items from senders? (and all the others as well?)

Reply:
The `MessageSender.close()` calls a helper method called `_deleteFromCache()` that removes the receiver from the context cache